Crypto/SecureDigest.cc

```diff
@@ -82,4 +87,49 @@ namespace litecore {
 #endif
     }
 
+#undef _CONTEXT
+#ifdef USE_COMMON_CRYPTO
+#include <CommonCrypto/CommonDigest.h>
+#define _CONTEXT ((CC_SHA256_CTX*)_context)
+#else
+#define _CONTEXT ((mbedtls_sha256_context*)_context)
+#endif
+
+    SHA256Builder::SHA256Builder() {
+        static_assert(sizeof(_context) >= sizeof(mbedtls_sha256_context));
+#ifdef USE_COMMON_CRYPTO
+        static_assert(sizeof(_context) >= sizeof(CC_SHA256_CTX));
+        CC_SHA256_Init(_CONTEXT);
+#else
+        mbedtls_sha256_init(_CONTEXT);
+        mbedtls_sha256_starts(_CONTEXT, 0);
+#endif
+    }
+
+
+    SHA256Builder& SHA256Builder::operator<< (fleece::slice s) {
+#ifdef USE_COMMON_CRYPTO
+        CC_SHA256_Update(_CONTEXT, s.buf, (CC_LONG)s.size);
+#else
+        mbedtls_sha256_update(_CONTEXT, (unsigned char*)s.buf, s.size);
+#endif
+        return *this;
+    }
+
+
+    void SHA256Builder::finish(void* result, size_t resultSize) {
+        DebugAssert(resultSize == sizeof(SHA256::_bytes));
+#ifdef USE_COMMON_CRYPTO
+        CC_SHA256_Final((uint8_t*)result, _CONTEXT);
+#else
+        mbedtls_sha256_finish(_CONTEXT, (uint8_t*)result);
+        mbedtls_sha256_free(_CONTEXT);
+#endif
+    }
+
+    // It seems odd to need these since they are base classes
```

Comment:
I think this happens because you've defined Hash::asBase64() outside the template declaration. Maybe? Templates are mysterious. But it's not a problem...